### PR TITLE
Various small chore fixes

### PR DESCRIPTION
**Pull request by fghaas**
These are just a handful of tiny patches that came up in preparing #124, but that can really land independently.

- refactor: Define macro for CC BY-SA
- feat: Add "Reporting issues" page
- refactor: Replace bare links to GitHub with non-bare ones

### CODE REVIEW

File: docs/contrib/issues.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Reporting issues
+
+If you notice an issue (a typo, a grammar glitch, or missing or incorrect information), here's how you can let us know:
```
Comment (colder-is-better) [inline docs/contrib/issues.md:3]: I would replace `or missing or incorrect information` with `missing or factually incorrect information`.

File: docs/contrib/issues.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+Go to [our issue tracker]({{config.repo_url}}/issues) and click *New issue*.
+
+![GitHub issues list with the "New issue" button shown in green](assets/issue-list.png)
+
+Select the issue type that most closely matches the type of issue that you are reporting.
```
Comment (colder-is-better) [inline docs/contrib/issues.md:9]: Personal preference: Drop `that`.

File: docs/contrib/issues.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+![GitHub "New issue" selection showing different types of issues](assets/issue-type-select.png)
+
+Be sure to fill *all* the required fields (the ones marked with a red asterisk), and fill the optional ones as necessary.
```
Comment (colder-is-better) [inline docs/contrib/issues.md:13]: Personal preference: Drop the `the` from `the required fields`, and replace `fill` with `fill in`.

File: docs/contrib/issues.md (hunk 1 above)
Comment (fghaas) [reply docs/contrib/issues.md:3]: Would you mind using [the suggestion feature](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request) to make your suggestions? I would like to try something out.

File: docs/contrib/issues.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+![GitHub "New issue" form for the "Missing or incomplete documentation" issue type](assets/issue-form.png)
+
+Don't forget to click *Submit new issue* when you're done.
+
+And if you want to help us **fix** the issue you've just reported, please take a look at our guidelines for [modifying content](modifications.md) on this site!
```
Comment (colder-is-better) [inline docs/contrib/issues.md:19]: Hey, you have preemptively dropped the `that` before `you've` :)

**Review by colder-is-better — COMMENTED** [review]
All is looking good! I only have some micro-nits.

File: docs/contrib/issues.md (hunk 1 above)
Comment (colder-is-better) [inline docs/contrib/issues.md:3]:
```suggestion
If you notice an issue (a typo, a grammar glitch, missing or factually incorrect information), here's how you can let us know:
```

Comment (colder-is-better) [reply docs/contrib/issues.md:3]: That's a really nice feature. I am updating!

File: docs/contrib/issues.md (hunk 2 above)
Comment (colder-is-better) [reply docs/contrib/issues.md:9]:
```suggestion
Select the issue type that most closely matches the type of issue you are reporting.
```

File: docs/contrib/issues.md (hunk 3 above)
Comment (colder-is-better) [reply docs/contrib/issues.md:13]:
```suggestion
Be sure to fill *all* required fields (the ones marked with a red asterisk), and fill in the optional ones as necessary.
```

**Review by colder-is-better — COMMENTED** [review]
I have some infinitesimally significant suggestions.

**Comment by fghaas** [comment]
All suggestions applied; please re-review and approve.

**Pull request merged**